lib/util/git/auth.ts

```diff
@@ -59,11 +59,9 @@ export function getGitAuthenticatedEnvironmentVariables(
     }
   }
 
-  const authenticationRules = getAuthenticationRulesWithToken(
-    originalGitUrl,
-    hostType,
-    token
-  );
+  const authenticationRules = token
+    ? getAuthenticationRulesWithToken(originalGitUrl, hostType, token)
+    : getAuthenticationRules(originalGitUrl, `${username}:${password}`);
```

Comment:
One question I had during implementation is if we should/need to URL encode the username/password here?

The token is not encoded, but the chance that some special URL characters like `/` `@` are part of the password is way more likely then in a token.

Reply:
probably yes, we need to add to sanitize after encoding too

Reply:
Added the encoding. 

While testing it I noticed that they are already getting correctly sanitized from the logs and I have no idea  yet why 😅

Reply:
Found it: 

https://github.com/renovatebot/renovate/blob/53263bbfda3eebf20506defd8ae78b9c53048766/lib/logger/utils.ts#L320-L324

Already removes all the username+password things from URLs